rclcpp/test/test_node.cpp

```diff
@@ -1377,6 +1376,7 @@ TEST_F(TestNode, set_parameters_undeclared_parameters_not_allowed) {
     EXPECT_EQ(node->get_parameter(name1).get_value<int>(), 2);
     EXPECT_EQ(node->get_parameter(name2).get_value<bool>(), true);  // old value
     EXPECT_EQ(node->get_parameter(name3).get_value<std::string>(), "red");
+    RCLCPP_SCOPE_EXIT({node->remove_on_set_parameters_callback(handler.get());});    // always reset
```

Comment:
I would move this directly after the callback is added above. This way an unexpected failure after calling `add_on_set_parameters_callback` won't prevent this remove function from being called. Also, having them close together makes it easier to associate them with each other.

Same comment applies to other places where the scoped exit macro is not next to the `add_on_set_parameters_callback` call.